tests/unit/test_loggingutil.py

```diff
@@ -245,3 +246,32 @@ def test_log_emit_and_glbl_cfg(
     # Check log emit does not access global config object:
     LOG.debug("Entering zero gravity")
     assert mock_cfg.get.call_args_list == []
+
+
+def test_patch_log_level(caplog: pytest.LogCaptureFixture):
+    """Test patch_log_level temporarily changes the log level."""
+    caplog.set_level(logging.DEBUG)
+    logger = logging.getLogger("forest")
+    assert logger.level == logging.NOTSET
+    logger.setLevel(logging.ERROR)
+    logger.info("nope")
+    assert not caplog.records
+    with patch_log_level(logger, logging.INFO):
+        LOG.info("yep")
+        assert len(caplog.records) == 1
+    logger.info("nope")
+    assert len(caplog.records) == 1
+
+
+def test_patch_log_level__reset(caplog: pytest.LogCaptureFixture):
```

Comment:
Why do these need to be separate tests?

Reply:
I'm not sure they could be (and even if they could, it would probably be very confusing). The above test is a test `patch_log_level()` does what it says. This test is a test that it is not affected by the root logger level. I'll add a bit more info to the docstring of this one